indexer/ai/src/main/java/com/google/cloudsearch/ai/AISkillBase.java

```diff
@@ -0,0 +1,123 @@
+            JSONArray propertyDefinitions = (JSONArray) obj.get("propertyDefinitions");
+            for(Object propertyObj : propertyDefinitions){
+                JSONObject property = (JSONObject) propertyObj;
+                String propertyName = (String) property.get("name");
+                schemaInfo.put(objName+"."+propertyName, true);
+            }
+        }
+        this.isSchemaInfoStored = true;
+    }
+
+    /**
+     * Helper function to validate property names in config
+     * @param propertyName  object name + property name specified in configuration
+     * @param schema    CloudSearch schema
+     * @return
+     */
+    private boolean isValid(String propertyName, JSONObject schema){
+
+        if(isSchemaInfoStored == false)
+            this.storeSchemaInfo(schema);
+        if(this.schemaInfo.get(propertyName) == null)
+            return false;
+        else
+            return true;
+    }
+
+    /**
+     * @param aiSkillName   Name of the AI Skill
+     * @throws InvalidConfigException   throws exception if name not specified in the configuration.
+     */
+    @Override
+    public void setAISkillName(String aiSkillName) throws InvalidConfigException {
+        if(aiSkillName == null || aiSkillName.equals(""))
+            throw new InvalidConfigException("No skill name specified.");
+
+        this.aiSkillName = aiSkillName;
+    }
+
+    /**
+     * @return  Returns skill name
+     */
+    @Override
+    public String getAISkillName() {
+        return this.aiSkillName;
+    }
+
+    /**
+     * @param outputMapping     JSON Array of Output Mappings
+     * @param schema            CloudSearch schema
+     * @throws InvalidConfigException   Throws exception if property name not present in schema
+     */
+    @Override
+    public void setOutputMappings(JSONArray outputMapping, JSONObject schema) throws InvalidConfigException {
+        if(outputMapping == null)
+            throw new InvalidConfigException("Output Mapping not specified.");
+
+        for(Object objMap : outputMapping){
+            JSONObject mappingObject = (JSONObject) objMap;
+            OutputMapping obj = new OutputMapping();
+            if(isValid((String) mappingObject.get(Constants.CONFIG_TARGET_PROPERTY), schema))
+                obj.setObjectAndPropertyName((String) mappingObject.get(Constants.CONFIG_TARGET_PROPERTY));
+            else
+                throw new InvalidConfigException("Invalid Object or Property Name");
+
+            obj.setSkillOutputField((String) mappingObject.get(Constants.CONFIG_OUTPUT_FILED));
+
+            String objectName = obj.getObjectName();
+            if(this.outputMappings.containsKey(objectName))
+                this.outputMappings.get(objectName).add(obj);
+            else {
+                List<OutputMapping> mapList  = new ArrayList<>();
+                mapList.add(obj);
+                this.outputMappings.put(objectName, mapList);
+            }
```

Comment:
Validation should not happen in setter, but in parse function.

Reply:
Yes, introduced new functions now.